From 114b6a383b02748dd1f811ff9ce4c2c91e41aeec Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Tue, 21 Nov 2017 22:30:34 +0100
Subject: [PATCH 2/4] rc: Fix broken python macros

This should address the broken python macros I submitted previously. I am terribly sorry about this!

Resolves #41

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 ypkg2/rc.yml | 32 ++++++++++++++++++++++++++++----
 1 file changed, 28 insertions(+), 4 deletions(-)

diff --git a/ypkg2/rc.yml b/ypkg2/rc.yml
index 4aaecf8..0394ffb 100644
--- a/ypkg2/rc.yml
+++ b/ypkg2/rc.yml
@@ -103,7 +103,13 @@ actions:
         perl_install
     - python_setup: |
         function python_setup() {
-            instdir=`basename $workdir`
+            if [[ -e $sources/.workdir ]]; then
+                cd $(cat $sources/.workdir)
+            else
+                echo $workdir > $sources/.workdir
+            fi
+
+            instdir=`basename $PWD`
             pushd ..
                 cp -a $instdir py2build && pushd py2build
                     python2.7 setup.py build $* || exit
@@ -113,7 +119,13 @@ actions:
         python_setup
     - python_install: |
         function python_install() {
-            instdir=`basename $workdir`
+            if [[ -e $sources/.workdir ]]; then
+                cd $(cat $sources/.workdir)
+            else
+                echo $workdir > $sources/.workdir
+            fi
+
+            instdir=`basename $PWD`
             pushd ..
                 if [[ ! -d py2build ]]; then
                     cp -a $instdir py2build
@@ -126,7 +138,13 @@ actions:
         python_install
     - python3_setup: |
         function python3_setup() {
-            instdir=`basename $workdir`
+            if [[ -e $sources/.workdir ]]; then
+                cd $(cat $sources/.workdir)
+            else
+                echo $workdir > $sources/.workdir
+            fi
+
+            instdir=`basename $PWD`
             pushd ..
                 cp -a $instdir py3build && pushd py3build
                     python3 setup.py build $* || exit
@@ -136,7 +154,13 @@ actions:
         python3_setup
     - python3_install: |
         function python3_install() {
-            instdir=`basename $workdir`
+            if [[ -e $sources/.workdir ]]; then
+                cd $(cat $sources/.workdir)
+            else
+                echo $workdir > $sources/.workdir
+            fi
+
+            instdir=`basename $PWD`
             pushd ..
                 if [[ ! -d py3build ]]; then
                     cp -a $instdir py3build
-- 
2.15.0

